src/lib/EventPublisher.js

```diff
@@ -61,37 +72,51 @@ class EventPublisher extends EventEmitter {
     };
 
     this.emit(eventData.name, event);
-    if (eventData.deviceID) {
-      this.emit(eventData.deviceID, event);
-      this.emit(`${eventData.deviceID}/${eventData.name}`, event);
-    }
-    this.emit("*all*", event);
+    this.emit(ALL_EVENTS, event);
   };
 
 
   subscribe = (
-    name: ?string,
+    eventName: ?string = ALL_EVENTS,
     eventHandler: (event: Event) => void,
-    deviceID: ?string,
+    filterOptions?: FilterOptions,
+    subscriberID?: string,
   ): void => {
     const subscriptionID = uuid();
-    const eventName = getEventName(name, deviceID);
-
-    this._subscriptionsByID.set(subscriptionID, { eventName, eventHandler });
-    this.on(eventName, eventHandler);
-
+    const listener = this._filterEvents(eventHandler, filterOptions);
+
+    this._subscriptionsByID.set(
+      subscriptionID,
+      {
+        listener,
+        eventName,
+        id: subscriptionID,
+        subscriberID,
+      },
+    );
+
+    this.on(eventName, listener);
     return subscriptionID;
   };
 
   unsubscribe = (subscriptionID: string): void => {
     const {
       eventName,
-      eventHandler,
+      listener,
     } = nullthrows(this._subscriptionsByID.get(subscriptionID));
 
-    this.removeListener(eventName, eventHandler);
+    this.removeListener(eventName, listener);
     this._subscriptionsByID.delete(subscriptionID);
   };
+
+  unsubscribeBySubscriberID = (subscriberID: string): void => {
```

Comment:
add subscriberID to be able to clean all listeners for the current device if it is disconnected.